app-modules/alert/src/Filament/Actions/BulkCreateAlertAction.php

```diff
@@ -0,0 +1,113 @@
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Alert\Filament\Actions;
+
+use AdvisingApp\Alert\Enums\AlertSeverity;
+use AdvisingApp\Alert\Enums\SystemAlertStatusClassification;
+use AdvisingApp\Alert\Models\AlertStatus;
+use AdvisingApp\Prospect\Models\Prospect;
+use AdvisingApp\StudentDataModel\Models\Student;
+use Exception;
+use Filament\Forms\Components\Select;
+use Filament\Forms\Components\Textarea;
+use Filament\Notifications\Notification;
+use Filament\Tables\Actions\BulkAction;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Support\Facades\DB;
+
+class BulkCreateAlertAction
+{
+    public static function make(): BulkAction
+    {
+        return BulkAction::make('createAlert')
+            ->label('Create Alert')
+            ->icon('heroicon-o-chat-bubble-bottom-center-text')
+            ->modalHeading('Create Alert')
+            ->form([
+                Textarea::make('description')
+                    ->required()
+                    ->string()
+                    ->label('Description'),
+                Select::make('severity')
+                    ->options(AlertSeverity::class)
+                    ->default(AlertSeverity::default())
+                    ->required()
+                    ->enum(AlertSeverity::class)
+                    ->label('Severity'),
+                Textarea::make('suggested_intervention')
+                    ->required()
+                    ->string()
+                    ->label('Suggested Intervention'),
+                Select::make('status_id')
+                    ->label('Status')
+                    ->options(AlertStatus::orderBy('order')->pluck('name', 'id'))
+                    ->default(fn () => SystemAlertStatusClassification::default()?->getKey())
+                    ->exists('alert_statuses', 'id')
+                    ->required(),
+            ])->action(function (Collection $records, array $data) {
+                try {
+                    DB::beginTransaction();
+
+                    $records->chunk(100)->each(function ($chunk) use ($data) {
```

Comment:
There is no need to chunk here since all the records are already loaded into memory